src/tests/unit/ingredients-dao.test.js

```diff
@@ -0,0 +1,145 @@
+import chai from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import config from 'config';
+import proxyquire from 'proxyquire';
+import sinon from 'sinon';
+import sinonChai from 'sinon-chai';
+
+import { getIngredientsData } from 'tests/unit/test-data';
+import { openapi } from 'utils/load-openapi';
+import { GetFilterProcessor } from 'utils/process-get-filters';
+
+const ingredientsGetParameters = openapi.paths['/ingredients'].get.parameters;
+
+sinon.restore();
+
+sinon.replace(config, 'get', () => ({ oracledb: {} }));
+
+chai.should();
+chai.use(chaiAsPromised);
+chai.use(sinonChai);
+
+describe('test ingredients dao', () => {
+  const {
+    getIngredientsQuery,
+    emptyGetIngredientsQuery,
+    testFilters,
+    ingredientAliases,
+    testBindParams,
+    testSerializerReturn,
+    testConditionals,
+    testConnectionReturn,
+    testConnectionReturnRows,
+  } = getIngredientsData;
+  let serializeIngredientsStub;
+  let getConnectionStub;
+  let connectionSpy;
+  let ingredientsDao;
+  let getFilterProcessorStub;
+  let getFilterProcessorContructorStub;
+  beforeEach(() => {
+    connectionSpy = sinon.stub().returns(testConnectionReturn);
+    serializeIngredientsStub = sinon.stub().returns(testSerializerReturn);
+    getConnectionStub = sinon.stub().resolves({
+      execute: connectionSpy,
+      close: () => {},
+    });
+    getFilterProcessorContructorStub = sinon.stub();
+    getFilterProcessorStub = sinon.stub(GetFilterProcessor.prototype, 'processGetFilters').returns({
+      bindParams: testBindParams,
+      conditionals: testConditionals,
+    });
+
+    Object.setPrototypeOf(GetFilterProcessor, getFilterProcessorContructorStub);
+
+    ingredientsDao = proxyquire('../../api/v1/db/oracledb/ingredients-dao', {
+      './connection': {
+        getConnection: getConnectionStub,
+      },
+      '../../serializers/ingredients-serializer': {
+        serializeIngredients: serializeIngredientsStub,
+      },
+    });
+  });
+  afterEach(() => {
+    sinon.restore();
+  });
+  describe('when the constructor for GetFilterProcessor is called', () => {
+    beforeEach(() => {
+      ingredientsDao = proxyquire('../../api/v1/db/oracledb/ingredients-dao', {
+        './connection': {
+          getConnection: getConnectionStub,
+        },
+        '../../serializers/ingredients-serializer': {
+          serializeIngredients: serializeIngredientsStub,
+        },
```

Comment:
Aren't these stubs already being taken care of in the top-level `beforeEach()`?

Reply:
left a comment about that [here](https://github.com/osu-mist/pizza-api/pull/9/files/873777b02cfaa8aaaa65bdbf84cd67d60d513171#r347666661). This `proxyquire` call is stubbing out `GetFilterProcessor` in addition to the normal stubs because we need to examine the interaction with the constructor, but it's left out in other tests because the constructor doesn't have side effects and it's much simpler to not stub it in those cases.